cpp_faster_fifo/tests/comparison_tests.py

```diff
@@ -0,0 +1,173 @@
+import multiprocessing
+from unittest import TestCase
+from faster_fifo import Queue
+from time import time
+from queue import Full, Empty
+import logging
+
+ch = logging.StreamHandler()
+ch.setLevel(logging.DEBUG)
+
+log = logging.getLogger('rl')
+log.setLevel(logging.DEBUG)
+log.handlers = []  # No duplicated handlers
+log.propagate = False  # workaround for duplicated logs in ipython
+log.addHandler(ch)
+
+MSG_SIZE = 5
+
+
+def make_msg(msg_idx):
+    return (msg_idx,) * MSG_SIZE
+
+
+def produce_msgs(q, p_idx, num_messages):
+    i = 0
+    while i < num_messages:
+        try:
+            q.put(make_msg(i), timeout=0.01)
+            if i % 100000 == 0:
+                log.info('Produce: %d %d', i, p_idx)
+            i += 1
+        except Full:
+            # time.sleep(0.001)
+            pass
+        except Exception as exc:
+            log.exception(exc)
+    log.info('Done producing! %d', p_idx)
+
+
+def consume_msgs(q, queue_type, p_idx, consume_many=None, total_num_messages=int(1e9)):
+    messages_received = 0
+    while True:
+        try:
+            msgs = q.get_many(timeout=0.01, max_messages_to_get=consume_many) if queue_type == 'ff' else q.get()
```

Comment:
I think you should write this differently. First of all, why use a string for the queue type? Just use the class itself.
Name it queue_cls, and then you can directly call it like queue_cls() to create a queue object. This is more elegant.

Second, one of the points of these tests is to show that our queue and multiprocessing.Queue can be used interchangeably. AND we have an extra option to get multiple messages with get_many(). So I would reorganize it like this:

if consume_many == 1:
    msg = q.get()
    msgs = [msg]
else:
    msgs = q.get_many(max_messages_to_get=consume_many)

This way you don't need to check for queue type also later when you calculate messages_received
Also, use 1 as default value for consume_many instead of 1. It works perfectly well semantically.

Reply:
Done